File: algorithm/src/test/java/programmers/level1/영단어변경/SolutionTest.java
package programmers.level1.영단어변경;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class SolutionTest {

    @Test
    void solution() {
        Solution solution = new Solution();

        // tc1
        int answer = solution.solution("one4seveneight");
        assertThat(answer).isEqualTo(1478);

        // tc2
        answer = solution.solution("23four5six7");
        assertThat(answer).isEqualTo(234567);

        // tc3
        answer = solution.solution("2three45sixseven");
        assertThat(answer).isEqualTo(234567);

        // tc4
        answer = solution.solution("123");
        assertThat(answer).isEqualTo(123);
    }
}

File: algorithm/src/test/kotlin/io/yomni/programmers/level1/87389/SolutionTest.kt
package io.yomni.programmers.level1.`87389`

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource

@DisplayName("SolutionTest")
class SolutionTest {

    companion object {
        @JvmStatic
        fun argumentsProvider() = listOf(
            Arguments.of(10, 3),
            Arguments.of(12, 11),
            Arguments.of(3, 2),
        )
    }

    @ParameterizedTest
    @MethodSource("argumentsProvider")
    fun solution(input: Int, expected: Int) {
        val solution = Solution()
        assertThat(solution.solution(input)).isEqualTo(expected)
    }
}
